pkg/jwt/configuration.go

```diff
@@ -102,3 +115,38 @@ func NewSignatureValidatorFromJSONWebKeySet(jwks *jose.JSONWebKeySet) (Signature
 
 	return NewDemultiplexingSignatureValidator(namedSignatureValidators, allSignatureValidators), nil
 }
+
+func NewSignatureValidatorFromJSONWebKeySetFile(path string, refreshInterval time.Duration) SignatureValidator {
+	// Hmm. I don't want us to first read the file once, to initialize this, and then read the file periodically to update it.
```

Comment:
Why wouldn't you want to do this? Doing a forced read upon startup seems like the most sensible thing to do. This ensures that if you get the pathname wrong in your config, your containers will crashloop and prevent a rolling upgrade to complete.

Reply:
It seemed a little inelegant to have two different code paths for the same thing, so my idea was to find some way to ensure that it would never be used before the first call to Replace(). I agree it does make sense to catch errors before we go into our goroutine, though. I've updated the code to do this.